File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/DonationIntentResult.java
package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

public class DonationIntentResult {
  @JsonProperty("id")
  private String id;

  @JsonProperty("client_secret")
  private String clientSecret;

  public DonationIntentResult(@JsonProperty("id") String id, @JsonProperty("client_secret") String clientSecret) {
    this.id           = id;
    this.clientSecret = clientSecret;
  }

  public String getId() {
    return id;
  }

  public String getClientSecret() {
    return clientSecret;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/DonationIntentResult.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

import com.fasterxml.jackson.annotation.JsonProperty

data class DonationIntentResult(
  val id: String,
  @JsonProperty("client_secret") val clientSecret: String
)
